python/lsst/dax/imgserv/api_soda.py

```diff
@@ -88,67 +88,81 @@ def load_imgserv_config(config_path=None, metaserv_url=None):
     # configure the log file (log4cxx)
     log.configure(os.path.join(config_path, "log.properties"))
     current_app.config["DAX_IMG_CONFIG"] = config_path
-    if metaserv_url is not None:
-        current_app.config["dax.imgserv.meta.url"] = metaserv_url
+    if metaserv_url:
+        current_app.config["DAX_IMG_META_URL"] = metaserv_url
     current_app.config["imgserv_api"]=os.path.join(config_path,
                                                    "image_api_schema.json")
     # create cache for butler instances
     current_app.butler_instances = {}
     # create SODA service
-    f_dashboard = os.path.join(config_path, "dashboard.json")
-    with open(f_dashboard, "r") as f:
-        dashboard = json.load(f)
-    current_app.soda = ImageSODA(current_app.config, dashboard)
+    current_app.soda = ImageSODA(current_app.config)
 
 
 @image_soda.route("/")
-def index():
+def is_index():
     return make_response(render_template("api_image_soda.html"))
 
 
 @image_soda.route("/availability", methods=["GET"])
-def imagesoda_availability():
+def is_availability():
     xml = current_app.soda.get_availability(_getparams())
     resp = make_response(xml)
-    resp.headers["Content_Type"] = "text/xml"
+    resp.headers["Content-Type"] = "text/xml"
     return resp
 
 
 @image_soda.route("/capabilities", methods=["GET"])
-def imagesoda_capabilities():
+def is_capabilities():
     xml = current_app.soda.get_capabilities(_getparams())
     resp = make_response(xml)
-    resp.headers["Content_Type"] = "text/xml"
+    resp.headers["Content-Type"] = "text/xml"
     return resp
 
 
 @image_soda.route("/examples", methods=["GET"])
-def imagesoda_examples():
+def is_examples():
     html = current_app.soda.get_examples(_getparams())
     return make_response(html)
 
 
+@image_soda.route("/tables", methods=["GET"])
+def is_tables():
+    xml = current_app.soda.get_tables(_getparams())
+    return make_response(xml)
+
+
+@image_soda.route("/sia", methods=["GET"])
+def is_sia():
+    resp = current_app.soda.do_sia(_getparams())
+    return make_response(resp)
+
+
 @image_soda.route("/sync", methods=["GET", "PUT", "POST"])
-def imagesoda_sync():
+def is_sync():
+    if not request.args:
+        # no parameters present, return service status
+        return _service_response(soda_method="sync")
     image = current_app.soda.do_sync(_getparams())
     if image:
         return _data_response(image)
     else:
         return _image_not_found()
 
 
-@image_soda.route("/async", methods=["POST"])
-def imagesoda_async():
-    params = _getparams()
-    xml = current_app.soda.do_async(params)
+@image_soda.route("/async", methods=["GET", "POST"])
+def is_async():
+    if not request.args:
+        # no parameters present, return service status
```

Comment:
Don't do this.  Just throw an error.  This is really not good because you aren't doing what you should be doing, but you'll return a 200?

Reply:
I was thinking the same earlier, until I checked out the reference SODA service (https://casda.csiro.au/casda_data_access/data/) in sodalint is returning, which is http 200 for /sync and /async (blank requests) along with the service descriptor.